src/audio/src/src_generic.c

```diff
@@ -113,8 +106,7 @@ static inline void fir_filter_generic(int32_t *rp, const void *cp, int32_t *wp0,
 		 */
 		data = fir_start + nch - j - 1;
 		for (i = 0; i < n2; i += nch) {
-			y0 += (int64_t)(*coef) * (*data);
-			coef++;
+			y0 += (int64_t)(*(coef++)) * (*data);
 			data += nch;
 		}
```

Comment:
and here
```
	for (i = 0; i < n2; i += nch, coef++, data += nch)
		y0 += (int64_t)*coef * data[0];
```

Reply:
Hmm, is it better? I don't like the "overloaded" for statement so much. I assume MCPS is not impacted.

Reply:
Some explanation - we will in some near future optimize these for Intel SSE, so I'd not put much effort to this version of code. The main purpose of it is to be clean looking reference code.

With the extra stuff in for () loop it's at least to me not so clear that all increments are "post" type.

Reply:
I agree with this, removed one more (), no performance impact, just looks better.